test/format/ops.c

```diff
@@ -525,68 +522,33 @@ begin_transaction(TINFO *tinfo, WT_SESSION *session, u_int *iso_configp)
 	*iso_configp = v;
 
 	/*
-	 * Keep trying to start a new transaction if it's timing out - we
-	 * know there aren't any resources pinned so it should succeed
-	 * eventually.
+	 * Lock out the oldest timestamp update and single-thread transaction
+	 * begin and prepare.
+	 */
+	testutil_check(pthread_rwlock_wrlock(&g.txn_lock));
```

Comment:
I only expected locking around allocating the timestamp and calling `timestamp_transaction`, not around the `begin_transaction`.  Does that not work for some reason?

Reply:
Agreed, pushed in 043fd79.